lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -787,16 +809,31 @@ test.serial('migrateGranulesAndFiles logs summary of migration for a specified l
   const {
     knex,
     testGranule,
+    testCollection,
   } = t.context;
-  process.env.loggingInterval = 1;
+
+  const testGranule2 = generateTestGranule({
+    collectionId: buildCollectionId(testCollection.name, testCollection.version),
+    execution: t.context.executionUrl,
+  });
 
   await granulesModel.create(testGranule);
+  await granulesModel.create(testGranule2);
 
   t.teardown(async () => {
     logSpy.restore();
     await granulesModel.delete(testGranule);
+    await granulesModel.delete(testGranule2);
   });
 
-  await migrateGranulesAndFiles(process.env, knex);
+  await migrateGranulesAndFiles(
+    process.env,
+    knex,
+    {
+      loggingInterval: 1,
+      parallelScanLimit: 1,
```

Comment:
I think for this test, given how this is mocked and what it states it's covering, we might need coverage for both scan/non-scan logic branches, although one *might* argue that could be set as a unit for `migrateGranuleAndFilesViaTransaction`.   There's a few options for resolution here.

Reply:
also.... more generally, because many of these tests are covering logic from migrateGranuleAndFilesViaTransaction, updating this for an if/else that *both* call `migrateGranuleAndFilesViaTransaction` eventually, we explicitly don't have coverage of those cases for the collection route.  

Meaning tests like `migrateGranulesAndFiles processes granule with no files` are really now `migrateGranulesAndFiles processes granule with no files when no granuleId or collectionId specified`

We *know* it's calling the same underlying code, but relying on that given we didn't write units for the underlying methods doesn't feel great.

Reply:
So we need an additional unit test to cover the logging interval behavior for a non-scan situation? 

Or more preferably, test this behavior in `migrateGranuleAndFilesViaTransaction` directly?

Just want to make sure I'm following.

Reply:
Yes, that... 'and'

The units that are covering `migrateGranulesAndFiles` are only testing the `scan` branch, so the cases that cover failures/etc (e.g. `processes all non-failing granule records and does not process files of failing granule records`) aren't testing that behavior for per-collection/"non scan" operations.  

While we know the underlying calls eventually resolve to the same logic, there's nothing ensuring that remains the case.   With top-down units like this that can create issues if we do a refactor. 

That said, given the maintenance cycle on this code, I'm inclined to not waste a ton of time chasing complete coverage on this via a refactor, but we're going to need to keep an eye on it in CR during phase 2/3.

Reply:
How about these changes?

825f51b63adeca5c5c128f147e8146610207cab4
c0c4167926d325cd84cf08c376b193d584ee92bc

Reply:
👍🏻  I think that approach addresses the concern sufficiently.    Thanks!